psl/diagnostics/src/collection.rs

```diff
@@ -62,6 +62,17 @@ impl Diagnostics {
 
         String::from_utf8_lossy(&message).into_owned()
     }
+
+    pub fn warnings_to_pretty_string(&self, file_name: &str, datamodel_string: &str) -> String {
+        let mut message: Vec<u8> = Vec::new();
+
+        for warn in self.warnings() {
+            warn.pretty_print(&mut message, file_name, datamodel_string)
+                .expect("printing datamodel warning");
+        }
+
+        String::from_utf8_lossy(&message).into_owned()
+    }
```

Comment:
This is used to snapshot the warnings in the new `introspection-tests` tests. At some point in the future, we might as well rename the existing `to_pretty_string` into `errors_to_pretty_string`, and create a new `to_pretty_string` method to be used in [`validation_tests.rs`](https://github.com/prisma/prisma-engines/blob/a03afcd23db6125a81a769b188e2af95679d2560/psl/psl/tests/validation_tests.rs#L16-L34).

Reply:
It will also be useful for https://github.com/prisma/prisma-private/issues/198